pandas/core/computation/expressions.py

```diff
@@ -133,7 +133,10 @@ def _evaluate_numexpr(op, op_str, a, b):
     roperator.rtruediv: "/",
     operator.floordiv: "//",
     roperator.rfloordiv: "//",
-    operator.mod: "%",
+    # we require Python semantics for mod of negative for backwards compatibility
```

Comment:
the relevant bug also affects rmod, floordiv, and rfloordiv

Reply:
have you got an example of floordiv, and rfloordiv failing.

```python
>>> pd.__version__
'1.1.2'
>>>
>>> from pandas.core.computation import expressions as expr
>>>
>>> expr.set_test_mode(True)
>>>
>>> s = pd.Series(np.arange(10001)) - 5000
>>> print(expr.get_test_result())
[True]
>>> result = s % 24
>>>
>>> print(expr.get_test_result())
[True]
>>>
>>> for i, elem in enumerate(s):
...     assert elem % 24 == result[i]
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AssertionError
>>>
>>> s = pd.Series(np.arange(10001)) - 5000
>>>
>>> print(expr.get_test_result())
[True]
>>> result = s % -24
>>> print(expr.get_test_result())
[True]
>>>
>>> for i, elem in enumerate(s):
...     assert elem % -24 == result[i]
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AssertionError
>>>
>>> s = pd.Series(np.arange(10001)) - 5000
>>> print(expr.get_test_result())
[True]
>>> s[s == 0] = 1
>>> print(expr.get_test_result())
[True]
>>> result = 24 % s
>>>
>>> print(expr.get_test_result())
[True]
>>>
>>> for i, elem in enumerate(s):
...     assert 24 % elem == result[i]
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AssertionError
>>>
>>>
>>> s = pd.Series(np.arange(10001)) - 5000
>>> print(expr.get_test_result())
[True]
>>>
>>> s[s == 0] = 1
>>> print(expr.get_test_result())
[True]
>>> result = -24 % s
>>>
>>> print(expr.get_test_result())
[True]
>>>
>>> for i, elem in enumerate(s):
...     assert -24 % elem == result[i]
...
Traceback (most recent call last):
  File "<stdin>", line 2, in <module>
AssertionError
>>>
>>>
>>> s = pd.Series(np.arange(10001)) - 5000
>>> print(expr.get_test_result())
[True]
>>>
>>> result = s // 24
>>>
>>> print(expr.get_test_result())
[]
>>>
>>> for i, elem in enumerate(s):
...     assert elem // 24 == result[i]
...
>>>
>>> result = s // -24
>>>
>>> print(expr.get_test_result())
[]
>>>
>>> for i, elem in enumerate(s):
...     assert elem // -24 == result[i]
...
>>>
>>> s = pd.Series(np.arange(10001)) - 5000
>>> print(expr.get_test_result())
[True]
>>> s[s == 0] = 1
>>> print(expr.get_test_result())
[True]
>>> result = 24 // s
>>>
>>> print(expr.get_test_result())
[]
>>>
>>> for i, elem in enumerate(s):
...     assert 24 // elem == result[i]
...
>>>
>>> result = -24 // s
>>>
>>> print(expr.get_test_result())
[]
>>>
>>> for i, elem in enumerate(s):
...     assert -24 // elem == result[i]
...
>>>
```

> the relevant bug...

I wasn't planning a bug fix, just a simple regression fix with revert of offending code. could open directly against 1.1.x?

Reply:
My bad, assumed it had the same behavior as cython without checking.

Reply:
it may do. not checked. but from the output above, numexpr not being used for floordiv, rfloordiv.

Is there special handling for that?

Reply:
In core.ops.array_ops.na_arithmetic_op we have a try/except for the numexpr call that catches TypeError.  Best guess is that floordiv _always_ gets caught here.